src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -61,6 +61,37 @@ public static bool IsZipDeployment(this IEnvironment environment)
                 !string.IsNullOrEmpty(environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteRunFromPackage));
         }
 
+        public static bool IsAppServiceWindowsEnvironment(this IEnvironment environment)
+        {
+            return environment.IsAppServiceEnvironment()
+                && !environment.IsLinuxAppServiceEnvironment()
+                && !environment.IsLinuxContainerEnvironment();
+        }
+
+        public static bool IsCoreToolsEnvironment(this IEnvironment environment)
+        {
+            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(CoreToolsEnvironment));
+        }
+
+        public static bool IsPersistentStorageAvailable(this IEnvironment environment)
```

Comment:
What kind of persistent storage are you talking about here? You mean file storage rather than a storage account? If so, I'd clarify the naming.

Reply:
From persistent storage i really mean the Environment supports downloading / storing something that would persist between restarts.